crates/swc_ecma_transforms_compat/src/es2015/parameters.rs

```diff
@@ -536,6 +564,29 @@ impl VisitMut for Params {
                             return_type: Default::default(),
                         },
                     });
+                    *e = match (self.in_prop, local_vars) {
+                        (true, Some(var_decl)) => Expr::Arrow(ArrowExpr {
```

Comment:
I think we should not generate an arrow function from here, as we apply `arrow` before `parameters`.

Reply:
A little bit context is needed. As discussed in #3268, Safari somehow still has bug for parameters until today(swc have `safari_id_destructuring_collision_in_function_expression` to fix it), so even if class properties is supported, a parameters pass is still needed. So this is only for Safari > 14.1 and `bugfixes: false`, where arrow function is supported perfectly.

[Babel](https://babeljs.io/repl#?browsers=safari%2014.1&build=&builtIns=false&corejs=3.6&spec=true&loose=false&code_lz=MYGwhgzhAECC0G8BQ1pmgXmgRmgamgApCA6MsASkwD40KVoAjTIskxqjW5ggFwAsAljAYReYXoODRpWQrJwAmAMyda0vkJEBfIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.17.2&externalPlugins=&assumptions=%7B%7D) behaves the same(you need to mannually close bugfixes), and there's no other way to capture `this` in class field

Reply:
But as this pass is applied after arrow, the emitted code will not work in browsers that only support es5.
I'm not talking about browers with support for arrows.

Reply:
For browser that doesn't support arrows, there's no chance they support class field, so this shall still work, but implicit dependency between passes is indeed a bad thing.

Reply:
Oh, I see.
My bad, sorry. I tried creating some example and found that class properties will not exist in ast in such cases. Thanks!